site/_posts/2020-04-27-sharing-custom-oauth-claims-with-a-asp-netcore-app.md

```diff
@@ -0,0 +1,95 @@
+
+{% include _image.liquid src="/assets/img/blogs/adding-more-claims-asp-net/aspnetextended-add-roles.png" alt="Adding roles to a FusionAuth application." class="img-fluid" figure=false %}
+
+Then we want to create a group. We'll call this the "ASP.NET Core User" group, and associate the user role with it.
+	
+{% include _image.liquid src="/assets/img/blogs/adding-more-claims-asp-net/aspnetextended-create-group.png" alt="Creating a group in a FusionAuth application." class="img-fluid" figure=false %}
+
+Finally, we need to add our user to our group. Navigate to the "newuser2@example.com" user (or any other user you've created and associated with the "dotnetcore" application) and go to the "Groups" tab. Add them to the "ASP.NET Core User" group. Warning: if a user isn't associated with an application, being associated with a group that has a role for an application does nothing.
+
+{% include _image.liquid src="/assets/img/blogs/adding-more-claims-asp-net/aspnetextended-add-user-to-group.png" alt="Adding our user to a group in a FusionAuth application." class="img-fluid" figure=false %}
+
+Let's login. As a reminder, here's how we can start the ASP.NET Core application (again [full source code](https://github.com/FusionAuth/fusionauth-example-asp-netcore) available here).
+
+```shell
+dotnet publish -r win-x64 && SampleApp__ClientSecret=H4... bin/Debug/netcoreapp3.1/win-x64/publish/SampleApp.exe
+```
+
+If you head to `http://localhost:5000` and click on the "Secure" page, you'll see the claims. If you were previously signed in, you may need to sign out and sign in again. The role is present in the JWT, as you can see below:
+
+{% include _image.liquid src="/assets/img/blogs/adding-more-claims-asp-net/aspnetextended-roles-only-roles-highlighted.png" alt="The secure page after a user has been associated with a role." class="img-fluid" figure=false %}
+
+Since the role and other claims are encoded in the JWT, which is stored in a cookie, whenever you change any user data in FusionAuth, you'll need log out and log in as that user to see the changes reflected in the ASP.NET Core application.
+
+## Adding custom claims
+
+You can add in custom claims based on user data, the time of day or anything else. In FusionAuth, you add this data to a JWT with a [lambda](https://fusionauth.io/docs/v1/tech/lambdas/jwt-populate). This is a JavaScript function which receives the JWT object before it is signed and can modify it. Remember way back when we [created the user](https://fusionauth.io/blog/2020/04/28/dot-net-command-line-client) and specified their favorite color as a command line option? Well, we're going to send that information down to the ASP.NET Core web application so that it can display that very valuable data.
```

Comment:
Delete "in" before custom claims
Change "which" to "that" - This is a JavaScript function that receives the JWT 
```suggestion
You can add custom claims based on user data,
```
```suggestion
This is a JavaScript function that receives the JWT object
```